src/main/python/hydro_sphere/deployment.py

```diff
@@ -207,6 +207,12 @@ def deploy(self, step):
             print("==> Running %s/%s script" % (dst_work_dir, script_name))
             self.run_cmd_on_multiple_instances(self.slave_instances, "/bin/bash " + dst_work_dir + "/" + script_name)
 
+        elif step == 17:
+            print ("==> Check that deployment is fine")
+            hydra_instance = self.master_instances[0]
+            hydra_instance.run_cmd("source ~/venv/bin/activate && cd " + dst_work_dir +
+                                   "/hydra-master && pyb run_integration_tests -x run_unit_tests")
```

Comment:
I tried to simulate the failure by creating an invalid hydra.ini file on hydra server. "pyb run_integration_test" stalled and did not return anything. Pipeline did NOT run any next step and failed because of timeout. 
As an other experiment, I create a return_failure.sh script on my master and called "bash return_failure.sh" rather than "pyb run_integration_tests". Pipeline stopped immediately by throwing error

 out: test.sh: line 3: return: -1: invalid option
[10.10.2.47] out: return: usage: return [n]
[10.10.2.47] out: 
Fatal error: run() received nonzero return code 2 while executing!
Requested: source ~/venv/bin/activate && cd /home/plumgrid/hydra-master && /bin/bash test.sh
Executed: /bin/bash -l -c "source ~/venv/bin/activate && cd /home/plumgrid/hydra-master && /bin/bash test.sh"

Fabric API automatically stops the script if it hits any error. That was one of the reasons, Fabric was chosen.

The change is safe to be merged. We may need to improve run_integration_test later though.